maven-extension/src/main/java/io/opentelemetry/maven/OpenTelemetrySdkService.java

```diff
@@ -89,60 +82,47 @@ public synchronized void dispose() {
+    if (logger.isDebugEnabled()) {
       logger.debug(
-          "OpenTelemetry: No -Dotel.exporter.otlp.endpoint property or OTEL_EXPORTER_OTLP_ENDPOINT "
-              + "environment variable found, use a NOOP OpenTelemetry SDK");
-    } else {
-      {
-        // Don't use a {@code io.opentelemetry.sdk.autoconfigure.spi.ResourceProvider} to inject
-        // Maven runtime attributes due to a classloading issue when loading the Maven OpenTelemetry
-        // extension as a pom.xml {@code <extension>}.
-        String initialCommaSeparatedAttributes =
-            OtelUtils.getSystemPropertyOrEnvironmentVariable(
-                "otel.resource.attributes", "OTEL_RESOURCE_ATTRIBUTES", "");
-        Map<String, String> attributes =
-            OtelUtils.getCommaSeparatedMap(initialCommaSeparatedAttributes);
-
-        // service.name
-        String serviceName =
-            OtelUtils.getSystemPropertyOrEnvironmentVariable(
-                "otel.service.name", "OTEL_SERVICE_NAME", null);
-
-        if (!attributes.containsKey(ResourceAttributes.SERVICE_NAME.getKey())
-            && StringUtils.isBlank(serviceName)) {
-          // service.name is not defined in passed configuration, we define it
-          attributes.put(
-              ResourceAttributes.SERVICE_NAME.getKey(),
-              MavenOtelSemanticAttributes.ServiceNameValues.SERVICE_NAME_VALUE);
-        }
-
-        // service.version
-        final String mavenVersion = this.runtimeInformation.getMavenVersion();
-        if (!attributes.containsKey(ResourceAttributes.SERVICE_VERSION.getKey())) {
-          attributes.put(ResourceAttributes.SERVICE_VERSION.getKey(), mavenVersion);
-        }
-
-        String newCommaSeparatedAttributes = OtelUtils.getCommaSeparatedString(attributes);
-        logger.debug(
-            "OpenTelemetry: Initial resource attributes: {}", initialCommaSeparatedAttributes);
-        logger.debug("OpenTelemetry: Use resource attributes: {}", newCommaSeparatedAttributes);
-        System.setProperty("otel.resource.attributes", newCommaSeparatedAttributes);
-      }
-
-      this.openTelemetrySdk = OpenTelemetrySdkAutoConfiguration.initialize(false);
-      this.openTelemetry = this.openTelemetrySdk;
+          "OpenTelemetry: OpenTelemetry SDK initialized with  "
+              + OtelUtils.prettyPrintSdkConfiguration(autoConfiguredOpenTelemetrySdk));
     }
+    this.openTelemetrySdk = autoConfiguredOpenTelemetrySdk.getOpenTelemetrySdk();
+    this.openTelemetry = this.openTelemetrySdk;
 
-    String mojosInstrumentationEnabledAsString =
-        System.getProperty(
-            "otel.instrumentation.maven.mojo.enabled",
-            System.getenv("OTEL_INSTRUMENTATION_MAVEN_MOJO_ENABLED"));
     this.mojosInstrumentationEnabled =
-        Boolean.parseBoolean(
-            StringUtils.defaultIfBlank(mojosInstrumentationEnabledAsString, "true"));
+        OtelUtils.getBooleanSysPropOrEnvVar("otel.instrumentation.maven.mojo.enabled")
```

Comment:
I think you should be able to use `getConfig(). get Boolean` on the autoconfigured SDK object

Reply:
Fixed